src/contracts/GPv2Settlement.sol

```diff
@@ -126,10 +126,32 @@ contract GPv2Settlement {
     }
 
     /// @dev Invalidate onchain an order that has been signed offline.
-    /// @param orderDigest The unique digest associated to the parameters of an
-    /// order. See [`orderUid`] for details.
-    function invalidateOrder(bytes32 orderDigest) public {
-        filledAmount[orderUid(orderDigest, msg.sender)] = uint256(-1);
+    /// @param orderUid The unique identifier of the order that is to be made
+    /// invalid after calling this function. The user that created the order
+    /// must be the the sender of this message. See [`extractOrderUidParams`]
+    /// for details on orderUid.
+    function invalidateOrder(bytes calldata orderUid) external {
+        (bytes32 orderDigest, uint32 validTo, address owner) =
+            extractOrderUidParams(orderUid);
+        require(owner == msg.sender, "GPv2: caller does not own order");
+        filledAmount[orderUidKey(orderDigest, validTo, msg.sender)] = uint256(
+            -1
+        );
+    }
+
+    /// @dev Return how much of the input order has been filled so far. See
+    /// [`filledAmount`] to know how this value depends on the order type.
+    /// @param orderUid The unique identifier associated to the order for which
+    /// to recover the filled amount. See [`extractOrderUidParams`] for details.
+    /// @return amount How much the order has been filled in absolute amount.
+    function getFilledAmount(bytes calldata orderUid)
+        external
+        view
+        returns (uint256 amount)
+    {
+        (bytes32 orderDigest, uint32 validTo, address owner) =
+            extractOrderUidParams(orderUid);
+        amount = filledAmount[orderUidKey(orderDigest, validTo, owner)];
```

Comment:
Nit: I don't know why nick changed the order in his last post, but ever before we had orderdigest, owner, validTo. This is also the order of the official [API documentation ](https://github.com/gnosis/gp-v2-services/pull/49)

Reply:
:see_no_evil: It was random every time I wrote it. I agree we should match the API and have `orderDigest || owner || validTo`.